CONTRIBUTING.md

```diff
@@ -1,6 +1,132 @@
+- **super( ) function:** Classes should use the built-in `super()` function when making calls to methods in their super-class(es) unless there are specific reasons not to. `super()` should be used consistently in all sub-classes since it does not work otherwise.
+- **Multiple Inheritance:** Multiple inheritance should be avoided in general without good reason.
+- **__init__.py:** The `__init__.py` files for modules should not contain any significant implementation code.  `__init__.py` can contain docstrings and code for organizing the module layout, however if a module is small enough that it fits in one file, it should simply be a single file, rather than a directory with an  `__init__.py` file.
+
+### Standard output, warnings, and errors
+
+- **Print Statement:** Used only for outputs in methods and scenarios explicitly requested by the user
+- **Errors and Exceptions:**  Always use the `raise` with built-in or custom exception classes. The nondescript `Exception` class should be avoided as much as possible, in favor of more specific exceptions (*IOError, ValueError* etc.).
+- **Warnings:**  Always use the `warnings.warn(message, warning_class)`for warnings. These get redirected to `log.warning()` by default, but one can still use the standard warning-catching mechanism and custom warning classes.
+- **Debugging and  Informational messages:**  Always use `log.info(message)` and `log.debug(message)`. The logging system uses the built-in Python logging module.
+
+### Data and Configuration
+
+- **Storing Data:**
+    - Packages can include data in a directory named *data* inside a subpackage source directory as long as it is less than about 100 kB.
+    - If the data exceeds this size, it should be hosted outside the source code repository, either at a third-party location on the internet.
+
+### Documentation and Testing
+
+- **Docstrings:**
+    - Docstrings must be provided for all public classes, methods, and functions.
+    - Docstrings should follow the [numpydoc style](https://numpydoc.readthedocs.io/en/latest/format.html) and reStructured Text format.
+    - Write usage examples in the docstrings of all classes and functions whenever possible. These examples should be short and simple to reproduce. Users should be able to copy them verbatim and run them.
+- **Unit tests:**  Provided for as many public methods and functions as possible, and should adhere to the standards set in the Testing Guidelines.
+- **Building Documentation:**
+    - Use sphinx to build the documentation.
+    - All extra documentation should go into a /docs sub-directory under the main stingray directory.
+
+## Testing Guidelines
+
+---
+
+The testing framework used by stingray is the pytest framework, it is accessed through the python [setup.py](http://setup.py/) test command. To run the tests, you will need to make sure you have the pytest package (version 3.1 or later) installed.
```

Comment:
Note that we are executing tests with `tox -e <test environment>`, after your APE 17 PR ;)

Reply:
Made changes